assets/src/components/fromThemeOptions/fromThemeOptions.js

```diff
@@ -45,16 +46,15 @@ const resolveDependency = ( theme, field, meta ) => {
 
   let dependencyValue = meta[ field.dependsOn ];
 
-  // The field has a dependency and the current value of the post is not in the list of options, so use default value of the dependency.
-  if ( dependencyConfiguration ) {
-    const dependencyValueIsAllowed = dependencyConfiguration.options && dependencyConfiguration.options.find( option => option.value === dependencyValue );
-    if ( !dependencyValueIsAllowed ) {
-      dependencyValue = dependencyConfiguration.default;
-    }
+  const dependencyValueIsAllowed = dependencyConfiguration.options?.find( option => option.value === dependencyValue );
+
```

Comment:
Removed this check as it's always true because of https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/669/files#diff-0a557c7b06bf5a384f50fa0e07bbb3e59ba232e788d9125e74ddad2f20ce013aL42